lua/nvim-tree/lib.lua

```diff
@@ -68,19 +69,25 @@ local function clone_node(node)
 end
 
 ---Api.tree.get_nodes
+---@return Node[]|nil
 function M.get_nodes()
   return clone_node(core.get_explorer())
 end
 
 -- If node is grouped, return the last node in the group. Otherwise, return the given node.
+---@param node Node
+---@return Node
 function M.get_last_group_node(node)
-  local next_node = node
-  while next_node.group_next do
-    next_node = next_node.group_next
+  while node and node.group_next do
+    node = node.group_next
   end
-  return next_node
+
+  ---@diagnostic disable-next-line: return-type-mismatch -- it can't be nil
```

Comment:
I can't see the possibility of a nil return given that the `node` parameter is not nil, or am I wrong?

Reply:
Looks like luals is getting confused.

We could refactor but this is fine.

We can revisit suppressions once we have types annotations present and correct.